staging/src/k8s.io/apiserver/plugin/pkg/audit/log/backend_test.go

```diff
@@ -159,8 +162,12 @@ func TestLogEventsJson(t *testing.T) {
 		},
 	} {
 		var buf bytes.Buffer
-		backend := NewBackend(&buf, FormatJson, auditv1beta1.SchemeGroupVersion)
-		backend.ProcessEvents(event)
+		backendIf := NewBackend(&buf, FormatJson, auditv1beta1.SchemeGroupVersion)
+		backendIf.ProcessEvents(event)
+		backendIf.Shutdown()
+		b := backendIf.(*backend)
+		b.worker()
```

Comment:
I'd like to see the asynchronous logic tested a little more thoroughly. The 3 things that could happen are: 1) process event 2) worker receives event 3) shutdown. I'd like to see some tests of those things happening in different orders.

Reply:
OK. will add some case